### PR TITLE
Add redirect-hostnames and remove redirect-alternates

### PR DESCRIPTION
**Pull request by dzcode**
Asana card: https://app.asana.com/0/1135662569808788/1135520579292304

Housekeeping:
- [x] Update `new sites and updates` and `existing sites` tabs of partner site google sheet to add `redirect-hostnames` and remove `redirect-alternates` field
    - [ ] For sites with `TRUE` in `redirect-alternates`, move their hostnames from the `alternate-hostnames` field to the `redirect-hostnames` field and add them to be rebuilt

- [x] Merge in changes to the import script in `turbovote-partner-sites`

~- [ ] Make changes in `user-works` or `turbovote-admin-http-api` so that searches/exports/stats for the main `domain` include old data from any `redirect-hostnames`~ Confirmed with Carmen that user data is migrated from the old hostname to the new one, so only appears under the new hostname -- meaning that search/export/stats in tv-admin won't need to be updated

- [ ] Confirm that the redirect URLs don't show up in TV-admin dropdowns

### CODE REVIEW

File: resources/schemas/20190814000-add-schema-deprecation-fields.edn [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[{:db/id #db/id [:db.part/db]
```
Comment (dzcode) [inline resources/schemas/20190814000-add-schema-deprecation-fields.edn]: @eggsyntax  is this what you had in mind?

**Comment by leilakass** [comment]
LGTM!

**Comment by teaforthecat** [comment]
Is the plan to remove :alternate-hostname/hostname from the code in the next release?

**Comment by dzcode** [comment]
> Is the plan to remove :alternate-hostname/hostname from the code in the next release?

Nope, it's going to stay, but be used only in specific cases. It will continue to be used for true alternate hostnames - e.g. some partner wants to have two domains (e.g. AARP has `aarp.turbovote.org` and `votingtool.aarp.org`, which don't redirect to each other) - but won't be used as it is currently for sites that have a deprecated domain that should redirect to a new one (e.g. going to `adu.turbovote.org` redirects to `ahu.turbovote.org`).

At the user level, I believe the difference is that a user who goes to `votingtool.aarp.org` has a `:hostname` stored as  `votingtool.aarp.org`, whereas a user that originally goes to `adu.turbovote.org` will have a hostname of `ahu.turbovote.org`

File: resources/schemas/20190814001-add-redirect-hostnames.edn [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ {:db/id #db/id [:db.part/db]
+  :db/ident :redirect-hostname/hostname
+  :db/valueType :db.type/string
+  :db/cardinality :db.cardinality/one
+  :db/doc "A previously used, deprecated hostname that should now be redirected."
+  :db.install/_attribute :db.part/db}
+
+ {:db/ident :partner/redirect-alternates
+  :schema/see-instead :partner/redirect-hostname
+  :schema/deprecated-at #inst "2019-08-16"
```
Comment (mrichards42) [inline resources/schemas/20190814001-add-redirect-hostnames.edn]: Do we need `deprecated-at`? The date is already on this schema file, and it's part of the transaction changing the schema. Not a big deal, just checking 🤷‍♂

File: src/partner_works/entities/partner_site.clj [hunk 3]
```diff
@@ -8,18 +8,22 @@
...
             [com.climate.newrelic.trace :refer [defn-traced]]))
 
-(defn hostname->tx-data [hostname]
-  {:db/id (db/tempid) :alternate-hostname/hostname hostname})
+(defn str->tx-data [field str]
+  "Takes a string and a datomic field key and turns it into tx data with a temp id"
+  {:db/id (db/tempid) field str})
 
-(defn hostnames->tx-data-coll [hostnames]
+(defn str-coll->tx-data-coll [field str-coll]
```
Comment (mrichards42) [inline src/partner_works/entities/partner_site.clj]: Nothing particular about strings here, so maybe this is `coll->tx-data`. And I realize this isn't your code, but these two functions are way more complicated than they need to be, vs:

```clj
(defn coll->tx-data [field coll]
  (set
   (for [v coll]
     {:db/id (db/tempid) field v})))
```

File: resources/schemas/20190814001-add-redirect-hostnames.edn (hunk 2 above)
Comment (dzcode) [reply resources/schemas/20190814001-add-redirect-hostnames.edn]: That's a good point, I'm fine removing it

**Pull request merged**